src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductVariantRepository.php

```diff
@@ -44,6 +45,24 @@ public function createQueryBuilderByProductCode(string $locale, string $productC
         ;
     }
 
+    public function createQueryBuilderByPhraseAndLocale(string $phrase, string $locale): QueryBuilder
+    {
+        $expr = $this->getEntityManager()->getExpressionBuilder();
+
+        return $this->createQueryBuilder('o')
+            ->innerJoin('o.translations', 'translation', 'WITH', 'translation.locale = :locale')
+            ->andWhere($expr->orX(
+                'translation.name LIKE :phrase',
+                'o.code LIKE :phrase'
+            ))
+            ->setParameter('phrase', '%' . $phrase . '%')
+            ->setParameter('locale', $locale)
+            ->orderBy('o.productId', 'ASC')
+            ->addOrderBy('o.position', 'ASC')
```

Comment:
This is the change in the behaviour - is it needed? Moreover, if we want to sort the results it could be done by `position` and `id` rather than `productId`

Reply:
No. `productId` is needed to control data order regardless of it's position is table. I checked it to be sure.

Reply:
It makes sense to me. You'll see variants of the same product together ordered the same way as everywhere in the admin or shop - by position.
However, this new column is redundant, since there's already a `product_id` column in that table.
You can replace it with `->orderBy('o.product', 'ASC')` and Doctrine will translate it to `ORDER BY o.product_id ASC`.